sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -721,16 +728,9 @@ public static <T> TypedRead<T> read(SerializableFunction<SchemaAndRecord, T> par
         .setDatumReaderFactory(
             (SerializableFunction<TableSchema, AvroSource.DatumReaderFactory<T>>)
                 input -> {
-                  try {
-                    String jsonTableSchema = BigQueryIO.JSON_FACTORY.toString(input);
-                    return (AvroSource.DatumReaderFactory<T>)
-                        (writer, reader) ->
-                            new GenericDatumTransformer<>(parseFn, jsonTableSchema, writer);
```

Comment:
Analyzing the code further, I saw that the original GenericDatumTransformer's constructor received a JSON string so that later it would re-parse that JSON into a TableSchema again but via a Supplier function.

Reply:
that was because TableSchema is not serializable